lib/OpenQA/WebAPI/Controller/API/V1/Comment.pm

```diff
@@ -77,11 +78,26 @@ text, date of update and the user name that created the comment.
 =cut
 
 sub list ($self) {
-    my $comments = $self->_comments();
-    return unless $comments;
+    my $comments = $self->_search_comments($self->_column_id);
+    return $self->render(json => {error => "No comments exist."}, status => 404) unless $comments;
     $self->render(json => [map { $_->extended_hash } $comments->all]);
 }
 
+sub _column_id ($self) {
+    return 'job_id' if $self->param('job_id');
+    return 'parent_group_id' if $self->param('parent_group_id');
+    return 'group_id';
+}
+
+sub _search_comments ($self, $param) {
```

Comment:
Not sure about this part.
`_comments` is used in other places as well, and we want a limit everywhere.
Maybe you can just extend `_obj_comments` to use [`search_related`](https://metacpan.org/pod/DBIx::Class::ResultSet#search_related).
e.g.
```perl
my $limit = ...;
my $obj = $self->app->schema->resultset($table)->find($id);
return $obj->search_related(comments => {}, {rows => $limit}) if $obj;
```